### PR TITLE
CLOUDP-68237: Fix test repo publishing

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-68237

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,7 +63,15 @@ make install
 Run `mongocli help` for a list of available commands
 or check our [online documentation](https://docs.mongodb.com/mongocli/master/) for more details.
 
-### Getting Started
+### Getting API Keys
```
Comment (gssbzn) [inline README.md:66]: sneaking this in

File: build/ci/evergreen.yml [hunk 2]
```diff
@@ -1156,15 +1156,15 @@ buildvariants:
...
+      - name: ".e2e .clusters .cloudmanager"
   - name: goreleaser_snapshot
     display_name: "Packaging and Publishing (goreleaser, apt, yum)"
     run_on:
       - macos-1014-codesign
     expansions:
       go_root: "/opt/golang/go1.14"
       go_bin: "/opt/golang/go1.14/bin"
-      server_version: "1.0.3"
+      server_version: "4.4.0-rc3"
```
Comment (gssbzn) [inline build/ci/evergreen.yml:1167]: we need to use an rc version here

File: build/ci/evergreen.yml [hunk 3]
```diff
@@ -1202,7 +1202,7 @@ buildvariants:
     run_on:
       - rhel70-small
     expansions:
-      server_version: "4.2.2"
+      server_version: "4.2.3"
```
Comment (gssbzn) [inline build/ci/evergreen.yml:1205]: prep for the next release

Comment (colm-quinn) [reply build/ci/evergreen.yml:1205]: This looks to be getting to the point where there's a number of steps to do for each release that we should track in a doc or .MD somewhere.

File: build/ci/evergreen.yml [hunk 4]
```diff
@@ -386,7 +386,7 @@ tasks:
           E2E_TAGS: atlas,config
   # If your e2e tests depend on a cluster running please consider setting it on its own task
   - name: atlas_generic_e2e
-    tags: ["e2e","generic"]
+    tags: ["e2e","generic","atlas"]
```
Comment (colm-quinn) [inline build/ci/evergreen.yml:389]: [q] How was this working before?

Comment (gssbzn) [reply build/ci/evergreen.yml:389]: it's duplicating if you check the waterfall atlas is running om tests and om is running atlas
hard to spot on the PR as we took them out of the PR suite and you'll only see it in the waterfall

File: build/ci/evergreen.yml (hunk 3 above)
Comment (gssbzn) [reply build/ci/evergreen.yml:1205]: Updated RELEASING.md

**Pull request merged**